packages/core/src/modules/oob/OutOfBandApi.ts

```diff
@@ -548,7 +548,7 @@ export class OutOfBandApi {
 
     await this.outOfBandService.updateState(this.agentContext, outOfBandRecord, OutOfBandState.PrepareResponse)
 
-    if (handshakeProtocols) {
+    if (handshakeProtocols && handshakeProtocols.length > 0) {
```

Comment:
Should we do the same with attached requests for consistency then?

Reply:
Good point!